tools/common/StaticRegistrar.cs

```diff
@@ -5251,6 +5596,87 @@ protected override bool SkipRegisterAssembly (AssemblyDefinition assembly)
 
 			return base.SkipRegisterAssembly (assembly);
 		}
+
+		// Find the value of the [UserDelegateType] attribute on the specified delegate
+		TypeReference GetUserDelegateType (TypeReference delegateType)
+		{
+			var delegateTypeDefinition = delegateType.Resolve ();
+			foreach (var attrib in delegateTypeDefinition.CustomAttributes) {
+				var attribType = attrib.AttributeType;
+				if (!attribType.Is (Namespaces.ObjCRuntime, "UserDelegateTypeAttribute"))
+					continue;
+				return attrib.ConstructorArguments [0].Value as TypeReference;
+			}
+			return null;
+		}
+
+		MethodDefinition GetDelegateInvoke (TypeReference delegateType)
+		{
+			var td = delegateType.Resolve ();
+			foreach (var method in td.Methods) {
+				if (method.Name == "Invoke")
+					return method;
+			}
+			return null;
+		}
+
+		MethodReference InflateMethod (TypeReference inflatedDeclaringType, MethodDefinition openMethod)
+		{
+			var git = inflatedDeclaringType as GenericInstanceType;
+			if (git is null)
+				return openMethod;
```

Comment:
nit: same can be achieved with a is not:
```suggestion
			if (nflatedDeclaringType is not GenericInstanceType  gitl)
				return openMethod;
```

Reply:
Fixed.